CI: Linux nightly job is failing due to mozjs build error.

Today's build: https://github.com/servo/servo/actions/runs/12426055359/job/34693769793
Yesterday's build: https://github.com/servo/servo/actions/runs/12406796536/job/34635831796

The file that fails to compile (pure_virtual.c) seems to be the same on we saw in some of the builds that were failing due to the sccache issue (#34571) but the error is different this time (a syntax error rather than an invalid `make` invocation):

```
2024-12-20T05:30:53.3290350Z   --- stderr
2024-12-20T05:30:53.3290724Z   cat: reconfigure.inputs: No such file or directory
2024-12-20T05:30:53.3291468Z   WARNING: When not building jemalloc, you need to build with --with-redist or set WIN32_REDIST_DIR.
2024-12-20T05:30:53.3292181Z   Reticulating splines...
2024-12-20T05:30:53.3292604Z   Finished reading 66 moz.build files in 0.11s
2024-12-20T05:30:53.3293192Z   Read 0 gyp files in parallel contributing 0.00s to total wall time
2024-12-20T05:30:53.3293822Z   Processed into 277 build config descriptors in 0.11s
2024-12-20T05:30:53.3294349Z   RecursiveMake backend executed in 0.21s
2024-12-20T05:30:53.3295037Z     180 total backend files; 180 created; 0 updated; 0 unchanged; 0 deleted; 6 -> 39 Makefile
2024-12-20T05:30:53.3295868Z   Total wall time: 0.49s; CPU time: 0.47s; Efficiency: 96%; Untracked: 0.06s
2024-12-20T05:30:53.3297040Z   In file included from /home/runner/.cargo/git/checkouts/mozjs-fa11ffc7d4f1cc2d/bb560b6/mozjs-sys/mozjs/build/pure_virtual/pure_virtual.c:5:
2024-12-20T05:30:53.3298914Z   /home/runner/work/servo/servo/target/production/build/mozjs_sys-887b0982c7a3a5b9/out/build/dist/include/mozilla/Assertions.h:327:3: error: expected identifier or '('
2024-12-20T05:30:53.3300171Z     MOZ_REALLY_CRASH(aLine);
2024-12-20T05:30:53.3300527Z     ^
2024-12-20T05:30:53.3301703Z   /home/runner/work/servo/servo/target/production/build/mozjs_sys-887b0982c7a3a5b9/out/build/dist/include/mozilla/Assertions.h:265:21: note: expanded from macro 'MOZ_REALLY_CRASH'
2024-12-20T05:30:53.3303077Z           MOZ_NOMERGE abort();                                        \
2024-12-20T05:30:53.3303568Z                       ^
2024-12-20T05:30:53.3304617Z   /home/runner/.cargo/git/checkouts/mozjs-fa11ffc7d4f1cc2d/bb560b6/mozjs-sys/mozjs/build/pure_virtual/pure_virtual.c:25:3: error: expected identifier or '('
2024-12-20T05:30:53.3305747Z     MOZ_CRASH("pure virtual call");
2024-12-20T05:30:53.3306116Z     ^
2024-12-20T05:30:53.3307232Z   /home/runner/work/servo/servo/target/production/build/mozjs_sys-887b0982c7a3a5b9/out/build/dist/include/mozilla/Assertions.h:296:7: note: expanded from macro 'MOZ_CRASH'
2024-12-20T05:30:53.3308603Z         MOZ_REALLY_CRASH(__LINE__);                                             \
2024-12-20T05:30:53.3309122Z         ^
2024-12-20T05:30:53.3310525Z   /home/runner/work/servo/servo/target/production/build/mozjs_sys-887b0982c7a3a5b9/out/build/dist/include/mozilla/Assertions.h:265:21: note: expanded from macro 'MOZ_REALLY_CRASH'
2024-12-20T05:30:53.3312225Z           MOZ_NOMERGE abort();                                        \
2024-12-20T05:30:53.3312701Z                       ^
2024-12-20T05:30:53.3313022Z   2 errors generated.
2024-12-20T05:30:53.3313900Z   make[4]: *** [/home/runner/.cargo/git/checkouts/mozjs-fa11ffc7d4f1cc2d/bb560b6/mozjs-sys/mozjs/config/rules.mk:600: pure_virtual.o] Error 1
2024-12-20T05:30:53.3315536Z   make[3]: *** [/home/runner/.cargo/git/checkouts/mozjs-fa11ffc7d4f1cc2d/bb560b6/mozjs-sys/mozjs/config/recurse.mk:72: build/pure_virtual/target-objects] Error 2
2024-12-20T05:30:53.3317147Z   make[2]: *** [/home/runner/.cargo/git/checkouts/mozjs-fa11ffc7d4f1cc2d/bb560b6/mozjs-sys/mozjs/config/recurse.mk:34: compile] Error 2
2024-12-20T05:30:53.3318560Z   make[1]: *** [/home/runner/.cargo/git/checkouts/mozjs-fa11ffc7d4f1cc2d/bb560b6/mozjs-sys/mozjs/config/rules.mk:361: default] Error 2
2024-12-20T05:30:53.3319805Z   make: *** [/home/runner/.cargo/git/checkouts/mozjs-fa11ffc7d4f1cc2d/bb560b6/mozjs-sys/makefile.cargo:145: all] Error 2
2024-12-20T05:30:53.3320977Z   thread 'main' panicked at /home/runner/.cargo/git/checkouts/mozjs-fa11ffc7d4f1cc2d/bb560b6/mozjs-sys/build.rs:345:5:
```

I tried with the [setup-clang](https://github.com/marketplace/actions/install-clang) action on my fork. This resolves the mozjs build issue, but now `mozangle` fails because bindgen seems to be using the older clang's lib path instead of clang-14 installed by the action. Exporting `LIBCLANG_PATH=/usr/lib/llvm-14/lib` doesn't seem to help either.

https://github.com/mukilan/servo/actions/runs/12429138152/job/34702391257

```
   /usr/lib/llvm-10/lib/clang/10.0.0/include/avx2intrin.h:41:21: error: use of undeclared identifier '__builtin_ia32_pabsd256'
  /usr/lib/llvm-10/lib/clang/10.0.0/include/avx2intrin.h:256:19: error: use of undeclared identifier '__builtin_ia32_pmaxsb256'
  fatal error: too many errors emitted, stopping now [-ferror-limit=]
  thread 'main' panicked at /home/runner/.cargo/registry/src/index.crates.io-6f17d22bba15001f/mozangle-0.5.3/build.rs:371:10:
  Should generate shader bindings: ClangDiagnostic("/usr/lib/llvm-10/lib/clang/10.0.0/include/emmintrin.h:2374:19: error: use of undeclared identifier '__builtin_ia32_pmaxsw128'\n/usr/lib/llvm-10/lib/clang/10.0.0/include/emmintrin.h:2394:19: error: use of undeclared identifier '__builtin_ia32_pmaxub128'\n/usr/lib/llvm-10/lib/clang/10.0.0/include/emmintrin.h:2414:19: error: use of undeclared identifier '__builtin_ia32_pminsw128'\n/usr/lib/llvm-10/lib/clang/10.0.0/include/emmintrin.h:2434:19: error: use of undeclared identifier '__builtin_ia32_pminub128'\n/usr/lib/llvm-10/lib/clang/10.0.0/include/tmmintrin.h:52:21: error: use of undeclared identifier '__builtin_ia32_pabsb128'\n/usr/lib/llvm-10/lib/clang/10.0.0/include/tmmintrin.h:88:21: error: use of undeclared identifier '__builtin_ia32_pabsw128'\n/usr/lib/llvm-10/lib/clang/10.0.0/include/tmmintrin.h:124:21: error: use of undeclared identifier '__builtin_ia32_pabsd128'\n/usr/lib/llvm-10/lib/clang/10.0.0/include/smmintrin.h:667:20: error: use of undeclared identifier '__builtin_ia32_pminsb128'\n/usr/lib/llvm-10/lib/clang/10.0.0/include/smmintrin.h:686:20: error: use of undeclared identifier '__builtin_ia32_pmaxsb128'\n/usr/lib/llvm-10/lib/clang/10.0.0/include/smmintrin.h:705:20: error: use of undeclared identifier '__builtin_ia32_pminuw128'\n/usr/lib/llvm-10/lib/clang/10.0.0/include/smmintrin.h:724:20: error: use of undeclared identifier '__builtin_ia32_pmaxuw128'\n/usr/lib/llvm-10/lib/clang/10.0.0/include/smmintrin.h:743:20: error: use of undeclared identifier '__builtin_ia32_pminsd128'\n/usr/lib/llvm-10/lib/clang/10.0.0/include/smmintrin.h:762:20: error: use of undeclared identifier '__builtin_ia32_pmaxsd128'\n/usr/lib/llvm-10/lib/clang/10.0.0/include/smmintrin.h:781:20: error: use of undeclared identifier '__builtin_ia32_pminud128'\n/usr/lib/llvm-10/lib/clang/10.0.0/include/smmintrin.h:800:20: error: use of undeclared identifier '__builtin_ia32_pmaxud128'\n/usr/lib/llvm-10/lib/clang/10.0.0/include/avx2intrin.h:29:21: error: use of undeclared identifier '__builtin_ia32_pabsb256'\n/usr/lib/llvm-10/lib/clang/10.0.0/include/avx2intrin.h:35:21: error: use of undeclared identifier '__builtin_ia32_pabsw256'\n/usr/lib/llvm-10/lib/clang/10.0.0/include/avx2intrin.h:41:21: error: use of undeclared identifier '__builtin_ia32_pabsd256'\n/usr/lib/llvm-10/lib/clang/10.0.0/include/avx2intrin.h:256:19: error: use of undeclared identifier '__builtin_ia32_pmaxsb256'\nfatal error: too many errors emitted, stopping now [-ferror-limit=]\n")
  stack backtrace:
     0: rust_begin_unwind
     1: core::panicking::panic_fmt
     2: core::result::unwrap_failed
     3: core::result::Result<T,E>::expect
     4: build_script_build::build_translator
     5: build_script_build::main
     6: core::ops::function::FnOnce::call_once
```

@sagudev do you have any idea why bindgen might be picking up the wrong libclang path even when setting the `LIBCLANG_PATH` environment variable?

--- a/.github/workflows/linux.yml
+++ b/.github/workflows/linux.yml
@@ -124,7 +124,14 @@ jobs:
         run: |
           git switch --detach
           git reset --hard FETCH_HEAD
-
+      - name: Install LLVM and Clang
+        # Expliclity install Clang 14 on Ubuntu 20.04 used by the nightly job (#34713).
+        if: ${{ runner.environment != 'self-hosted' && inputs.upload }}
+        uses: KyleMayes/install-llvm-action@v2
+        with:
+          version: 14
+        # FIXME: It would be better to use the above step for regular Linux jobs
+        # and remove the following step, but currenty that is failing and needs investigation.
       - name: Set LIBCLANG_PATH env # needed for bindgen in mozangle
         if: ${{ runner.environment != 'self-hosted' && !inputs.upload }} # not needed on ubuntu 20.04 used for nightly
         run: echo "LIBCLANG_PATH=/usr/lib/llvm-14/lib" >> $GITHUB_ENV

